data/campaigns/Descent_Into_Darkness/scenarios/03_A_Haunting_in_Winter.cfg

```diff
@@ -132,222 +111,986 @@
-            [have_unit]
-                side=4
-                canrecruit=yes
-            [/have_unit]
-            [variable]
-                name=turn_number
-                greater_than=$free_ghost_turn
-            [/variable]
-        [/filter_condition]
-
-        {MODIFY_ONE_UNIT (side,race=1,undead) (side=4)}
+        name=last breath
+        [filter]
+            id=S3_Rebel_Soul
+        [/filter]
+
+        [message]
+            speaker=Darken Volk
+            message=_ "Seize control back while it is weakened!"
+        [/message]
+
+        [scroll_to]
+            x,y=$x1,$y1
+        [/scroll_to]
+
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 1}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 2}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 1}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 2}
+        [delay]
+            time=400
+        [/delay]
+        {MODIFY_UNIT (id=S3_Rebel_Soul) side 1}
+        [delay]
+            time=400
+        [/delay]
+
+        [message]
+            speaker=Malin Keshar
+            message= _ "I think I’ve got it under control again. That was a close call. Let me try the other one now."
+        [/message]
+
+        [animate_unit]
+            [filter]
+                id=Malin Keshar
+            [/filter]
+            [primary_attack]
+                range=ranged
+            [/primary_attack]
+            flag=attack
+        [/animate_unit]
+
+        [delay]
+            time=400
+        [/delay]
+
+        [unit]
+            type=Ghost
+            x,y=$loc2.x,$loc2.y
+            side=1
+            animate=yes
+        [/unit]
+
+        [delay]
+            time=500
+        [/delay]
+
+        [message]
+            speaker=Darken Volk
+            message= _ "No mishaps this time, but you will need to... practice more."
+        [/message]
+
+        {CLEAR_VARIABLE loc1}
+        {CLEAR_VARIABLE loc2}
+
+        [terrain]
+            x=11,12
+            y=3,3
+            terrain=Re
+        [/terrain]
+        [terrain]
+            x=11,12
+            y=4,4
+            terrain=Uu
+        [/terrain]
+        [redraw][/redraw]
+
+        {FULL_HEAL (side=1)}
+
+        {OBJECTIVE_EXPLORE}
     [/event]
 
+    # Fifth area (scorpions/trolls)
     [event]
-        name=die
+        name=enter hex
         [filter]
-            side=4
-            canrecruit=yes
+            side=1
+            x,y=5-9,6-7
         [/filter]
 
-        # Regain control of "rebel" ghosts
-        {MODIFY_UNIT (side,canrecruit=4,no) side 1}
-        [redraw]
-            clear_shroud=yes
-        [/redraw]
+        [message]
+            speaker=Malin Keshar
+            message= _ "Rail tracks and mines... dwarves used to live here, but there do not seem to be any guards. I wonder where they went?"
```

Comment:
Should there be a change to the .map file in this branch?

There don't seem to be any rail tracks or mine-walls on the map. Several monsters seem to spawn inside walls, and the visibility seems wrong (the young ogres at the start are shrouded).

Reply:
Yes, I totally forgot to push the new map! D:

I'll do that tonight when I get back to my other computer.